src/Pages/AdminTaskTable/components/ViewPayloadModal/ViewPayloadModal.test.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  defaultIntrospectTask,
+  defaultSnapshotTask,
+  ReactQueryTestWrapper,
+} from '../../../../testingHelpers';
+import ViewPayloadModal from './ViewPayloadModal';
+import { fireEvent, render } from '@testing-library/react';
+import { AdminTask } from '../../../../services/AdminTasks/AdminTaskApi';
+
+jest.mock('../../../../services/Notifications/Notifications', () => ({
+  useNotification: () => ({ notify: () => null }),
+}));
+
+jest.mock('../../../../services/useDebounce', () => (value) => value);
+jest.mock('../../../../middleware/AppContext', () => ({ useAppContext: () => ({}) }));
+
+it('Render loading spinner', async () => {
+  render(
+    <ReactQueryTestWrapper>
+      <ViewPayloadModal open isLoading={true} adminTask={null} setClosed={() => undefined} />
+    </ReactQueryTestWrapper>,
+  );
+});
+
+it('Open introspect task details and click tabs', async () => {
+  const { queryByText } = render(
+    <ReactQueryTestWrapper>
+      <ViewPayloadModal
+        open
+        isLoading={false}
+        adminTask={defaultIntrospectTask}
+        setClosed={() => undefined}
+      />
+    </ReactQueryTestWrapper>,
+  );
+
+  expect(queryByText('2375c35b-a67a-4ac2-a989-21139433c172')).toBeInTheDocument();
+
+  const detailsTab = queryByText('Task details');
+  expect(detailsTab).toBeInTheDocument();
+  const detailsContent = document.getElementById('task-details');
+  expect(detailsContent).toBeInTheDocument();
+  expect(detailsContent).not.toHaveAttribute('hidden');
+
+  const payloadTab = queryByText('Payload') as Element;
+  expect(payloadTab).toBeInTheDocument();
+  const payloadContent = document.getElementById('Payload');
+  expect(payloadContent).toBeInTheDocument();
+  expect(payloadContent).toHaveAttribute('hidden');
+
+  fireEvent.click(payloadTab);
+  expect(detailsContent).toHaveAttribute('hidden');
+  expect(payloadContent).not.toHaveAttribute('hidden');
+});
+
+it('Open snapshot details and click tabs', async () => {
+  const { queryByText } = render(
+    <ReactQueryTestWrapper>
+      <ViewPayloadModal
+        open
+        isLoading={false}
+        adminTask={defaultSnapshotTask}
+        setClosed={() => undefined}
+      />
+    </ReactQueryTestWrapper>,
+  );
+
+  expect(queryByText('2375c35b-a67a-4ac2-a989-21139433c172')).toBeInTheDocument();
```

Comment:
Same here, if the core of these values come from "defaultSnapshotTask", us the destructured defaultSnapshotTask to obtain stable variables here.